x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -0,0 +1,368 @@
+            public void onResponse(GetPipelineResponse getPipelineResponse) {
+                if (getPipelineResponse.isFound() == false) {
+                    createPipeline(client);
+                } else {
+                    logger.info("Found " + ENT_SEARCH_INGESTION_PIPELINE_NAME + " pipeline.");
+                }
+            }
+
+            public void onFailure(Exception e) {
+                final Throwable cause = ExceptionsHelper.unwrapCause(e);
+                if (cause instanceof ResourceNotFoundException) {
+                    logger.error("Pipeline " + ENT_SEARCH_INGESTION_PIPELINE_NAME + " not found.");
+                    createPipeline(client);
+                } else {
+                    logger.error("Error getting " + ENT_SEARCH_INGESTION_PIPELINE_NAME + " index " + e.toString());
+                }
+            }
+        }, client.admin().cluster()::getPipeline);
+    }
+
+    private static Settings getConnectorsIndexSettings() {
+        return Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-3")
+            .put(IndexMetadata.SETTING_INDEX_HIDDEN, true)
+            .put("index.refresh_interval", "1s")
+            .build();
+    }
+
+    private static XContentBuilder getConnectorsIndexMappings() {
+        try {
+            final XContentBuilder builder = jsonBuilder();
+            builder.startObject();
+            {
+                builder.startObject("_meta");
+                builder.field("version", Version.CURRENT.toString());
+                builder.endObject();
+
+                builder.field("dynamic", "false");
+                builder.startObject("properties");
+                {
+                    builder.startObject(Connector.ERROR_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.INDEX_NAME_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.IS_NATIVE_FIELD.getPreferredName());
+                    builder.field("type", "boolean");
+                    builder.endObject();
+
+                    builder.startObject(Connector.LANGUAGE_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.LAST_SEEN_FIELD.getPreferredName());
+                    builder.field("type", "date");
+                    builder.endObject();
+
+                    builder.startObject(Connector.LAST_SYNC_ERROR_FIELD.getPreferredName());
+                    builder.field("type", "text");
```

Comment:
This should be `keyword`: https://github.com/elastic/ent-search/blob/main/shared_togo/lib/shared_togo/crawler2/connector.rb#L135

Reply:
Hmm, that seems wrong tbh. This field will contain a full-text error message, should that really be a keyword?

Reply:
Ah, wait, now I remember. Nevermind!